### ISSUE TITLE
Unit + Acceptance tests

### ISSUE THREAD

**Issue by pamtbaau** [hughbris/grav-plugin-pushy#17]
Do you consider Unit and Acceptance tests as beneficial or an overhead? If you consider them as an asset, I'll create them.

**Comment by hughbris** [comment]
Please do and tell me how to run them.

I don't know if that would include tests on a deployment setup. I think with so much to configure in this plugin, that would be invaluable. Perhaps that's better suited to a tool within the plugin.

**Comment by pamtbaau** [comment]
> I don't know if that would include tests on a deployment setup.

No, it tests if the software written is working as expected. Unit testing is testing small parts of the software, eg. methods, while Acceptance is mimicking a person (your client) hammering Admin through the browser.

Creating shell scripts might be handy to automate the setup of a local/remote machine.

**Comment by hughbris** [comment]
Thanks. I think down the track, in a plugin with so many pieces to set up, it would be useful to add helpers and test buttons for some of its parts in the Admin UI.

**Comment by pamtbaau** [comment]
I've create a few Unit tests for RequestHandler (not so much to test though...) and started with creating Acceptance tests. 

Since the Admin UI has been build with JavaScript, a real browser must be used to run the tests. Codeception only supports Google Chrome and Firefox.

I'm using Chrome, but I do remember that you are not a fan of Google and probably also not its browser...

On the other hand, I wonder if Acceptance tests are really needed. There is not much of a UI to test. Manual testing if the UI looks good and dirty pages are being fetched and can be published doesn't seem to be a big burden that must be automated.

Testing the deployment of it all might be a different story, but doesn't require UI testing if I interpret your plans correctly.

For the moment, I've put Acceptance testing on hold.
